app/javascript/react/screens/App/Overview/OverviewValidators.js

```diff
@@ -0,0 +1,222 @@
+import { array, string, object } from 'yup';
+import { logSchemaError } from '../../../../common/schemaHelpers';
+
+const providersSchema = array()
```

Comment:
for some reason I got this itch... that all schemas should be in their own file... kinda like how we have constants in their own file... 

(BTW writing this comment after the first one, skip down to the one on line 14 then continue reading)

And if there exists a generic validator function stored in some `common` folder ☝️ becomes a real real reality

Reply:
i started it this way... but then i realized we query different attributes in different locations. I can look into extending a base schema object then?

Reply:
So I did some more research on this - it looks it's possible to achieve what we want using `schema.concat()` (so we can extend arrays/objects/etc.), but we just have to keep in mind that once a base type has been defined for a field, it must be compatible once extended. Here is a simple example that demonstrates this:

```
const contactSchema = object({
  name: string()
    .required(),
  email: string()
    .email(),
  address: object({
    line1: string().required(),
    zipcode: string().required()
  })
})
```
can be extended like this (say you want to extend address and add line2 after the fact):
```
let contactAddress = object({
    address: object({
      line2: string().required()
  })
});

let contactSchemaEnhanced = contactSchema.concat(contactAddress);
let contact = {
  name: 'jimmy',
  email: 'jdog@cool.biz',
  address: {
    zipcode: '12345'
  }
}
await contactSchemaEnhanced.validate(contact, { abortEarly: false, strict: true })
```
and this would validate and work like you would expect (giving line1 and line2 validation errors)...

*However*, two schemas are considered incompatible if the overrides for the same field are not the same type... so extending like this:
```
let contactAddress = object({
    address: object({
     line2: string().required(),
     zipcode: number().required()
  })
});
let contactSchemaEnhanced = contactSchema.concat(contactAddress);
```
would give an error: 
`You cannot `concat()` schema's of different types: string and number`

This is pretty much consistent with our data model (you'd never expect the same field to return w/ two different types to my knowledge). In the cases that does happen, we can look at `when()` and `if()` for dynamic type validation (but those should be rare).

https://github.com/jquense/yup#mixedwhenkeys-string--arraystring-builder-object--value-schema-schema-schema

So this being said, I will update these current schemas to use a base provider, cluster, datastore, lan, plan, request, etc. and we can adjust these as needed as we find different variations in the models coming back... (I think alongside the reducer in the `{Reducer}Validations.js` file is still probably a good place to add specific overrides.